docs/release_notes.md

```diff
@@ -1,3 +1,327 @@
+## 4.2.9
+
+Release Date: 2021-05-xx
+
+### Enhancements
+
+ - Add rule priority to delayed rules [#2759]
+
+ - Move delay server context into database [#3049] [#4428] [#5153]
+
+ - Add tests for partial copying via istream [#4698]
+
+ - Replication to hierarchies [#3217] [#4686]
+
+ - Refactor iphymv [#3490] [#4212] [#4563] [#4896] [#5070] [#5177] [#5360] [#5446] [#5454]
+
+ - New msiTouch, itouch, and touch API plugin [#4669] [#4694] [#5152]
+
+ - Refactor checksum API [#3091] [#3282] [#3540] [#5251] [#5252] [#5263] [#5285] [#5288] [#5400] [#5401] [#5496]
+
+ - Add SHA1, SHA512, ADLER32, hashing functions [#3800] [#5392]
+
+ - Add logical locking [#1781] [#3848] [#4170] [#4236] [#4433] [#4958] [#5312] [#5421] [#5486] [#5495] [#5542]
+
+ - Add client support for updating mtime of collections [#4190]
+
+ - New finalize_data_object API plugin [#2719] [#4331]
```

Comment:
API is actually called data_object_finalize

Reply:
done